src/transaction/utils.ts

```diff
@@ -412,7 +417,136 @@ function convertMemo(memo: Memo): { Memo: ApiMemo } {
   };
 }
 
+/**
+ * Autofills fields in a transaction.
+ *
+ * @param {Transaction} tx A transaction to autofill fields.
+ * @param {Client} client A client.
+ * @returns {Transaction} An autofilled transaction.
+ */
+async function autofillTransaction(
+  tx: Transaction,
+  client: Client
+): Promise<Transaction> {
+  validateAccountAddress(tx, 'Account', 'SourceTag');
+  if (tx['Destination']) {
+    validateAccountAddress(tx, 'Destination', 'DestinationTag');
+  }
+
+  // DepositPreauth:
+  convertToClassicAddress(tx, 'Authorize');
+  convertToClassicAddress(tx, 'Unauthorize');
+  // EscrowCancel, EscrowFinish:
+  convertToClassicAddress(tx, 'Owner');
+  // SetRegularKey:
+  convertToClassicAddress(tx, 'RegularKey');
+
+  setTransactionFlagsToNumber(tx);
+
+  if (tx.Sequence === undefined) {
+    const sequence = await getNextValidSequenceNumber(tx.Account, client);
+    tx.Sequence = sequence;
+  }
+
+  if (tx.Fee === undefined) {
+    tx.Fee = await calculateFeePerTransactionType(tx, client);
+  }
+
+  if (tx.LastLedgerSequence === undefined) {
+    const ledgerSequence = await getLatestValidatedLedgerSequence(client);
+    tx.LastLedgerSequence = ledgerSequence + LEDGER_OFFSET;
+  }
+
+  return tx;
+}
+
+function validateAccountAddress(tx: Transaction, accountField: string, tagField: string): void {
+  // if X-address is given, convert it to classic address
+  const { classicAccount, tag } = getClassicAccountAndTag(tx[accountField]);
+  tx[accountField] = classicAccount;
+
+  if (tag !== null) {
+    if (tx[tagField] && tx[tagField] !== tag) {
+      throw new ValidationError(
+        `The ${tagField}, if present, must match the tag of the ${accountField} X-address`
+      );
+    }
+    tx[tagField] = tag;
+  }
+}
+
+function convertToClassicAddress(tx: Transaction, fieldName: string): void {
+  const account = tx[fieldName];
+  if (typeof account === "string") {
+    const { classicAccount } = getClassicAccountAndTag(account);
+    tx[fieldName] = classicAccount;
+  }
+}
+
+async function getNextValidSequenceNumber(
+  account: string,
+  client: Client
+): Promise<number> {
+  const request: AccountInfoRequest = {
+    command: 'account_info',
+    account,
+  };
+  const data = await client.request(request);
+  return data.result.account_data.Sequence;
+}
+
+async function calculateFeePerTransactionType(
+  transaction: Transaction,
+  client?: Client
+): Promise<string> {
+  let netFee;
+  if (client === undefined) {
+    netFee = 10; // 10 drops
+  } else {
+    netFee = await client.getFee(); // Usually 0.00001 XRP (10 drops)
+  }
+
+  let baseFee = netFee;
```

Comment:
You might need to do some more processing on this - https://github.com/ripple/ripple-lib/blob/develop/src/transaction/utils.ts#L278